src/RendezvousTracker.ts

```diff
@@ -0,0 +1,268 @@
+                fileNameAsBrs = this.convertDebuggerPathToClient(fileNameAsBrs);
+                if (fileNameAsBrs) {
+                    fileName = fileNameAsBrs;
+                } else {
+                    fileNameAsXml = this.convertDebuggerPathToClient(fileNameAsXml);
+                    if (fileNameAsXml) {
+                        fileName = fileNameAsXml;
+                    }
+                }
+            }
+
+            this.clientPathsMap[fileName] = {
+                clientPath: this.convertDebuggerPathToClient(fileName),
+                clientLines: {
+                    [lineNumber]: this.convertDebuggerLineToClientLine(fileName, parseInt(lineNumber))
+                }
+            };
+        } else if (!this.clientPathsMap[fileName].clientLines[lineNumber]) {
+            // Add new client line to clint path map
+            this.clientPathsMap[fileName].clientLines[lineNumber] = this.convertDebuggerLineToClientLine(fileName, parseInt(lineNumber));
+        }
+
+        return fileName;
+    }
+
+    /**
+     * Helper function to assist in the creation of a RendezvousLineInfo
+     * @param fileName processed file name
+     * @param lineNumber occurrence line number
+     * @param duration how long the rendezvous took to complete, if not supplied it is assumed to be zero
+     */
+    private createLineObject(fileName: string, lineNumber: number, duration?: string): RendezvousLineInfo {
+        return {
+            clientLineNumber: lineNumber,
+            clientPath: this.clientPathsMap[fileName].clientPath,
+            hitCount: 1,
+            totalTime: this.getTime(duration),
+            type: 'lineInfo'
+        };
+    }
+
+    /**
+     * Helper function to convert the duration to a float or return 0.00
+     * @param duration how long the rendezvous took to complete, if not supplied it is assumed to be zero
+     */
+    private getTime(duration?: string): number {
+        return duration ? parseFloat(duration) : 0.000;
+    }
+
+    /**
+     * Replaces the last occurrence of a string in a string
+     * @param base String to search within
+     * @param replace String to replace
+     * @param replacement What to replace the final occurrence with
+     */
+    private replaceLastStringOccurrence(base: string, replace: string, replacement: string): string {
+        return (base.includes(replace)) ? base.substr(0, base.lastIndexOf(replace)) + replacement : base;
+    }
+}
+
+/**
+ * Used to see if a field on a object in RendezvousHistory object is a reserved field
+ * @param fieldName name of the field to check
+ */
+export function isRendezvousDetailsField(fieldName: string): boolean {
+    return (fieldName === 'type' || fieldName === 'hitCount' || fieldName === 'totalTime' || fieldName === 'zeroCostHitCount');
+}
+
+export interface RendezvousHistory {
+    [key: string]: RendezvousFileInfo;
+}
+
+interface RendezvousFileInfo {
+    [key: string]: RendezvousLineInfo | ElementType | number;
```

Comment:
Don't make this object indexable like this. From what I can tell, the indexable item is the line information, so instead, add a property to this interface (and the actual objects where created) called `lines` or something. 

Also, since you are the one constructing this object, each property should be a single object type (instead of the three that it is right now). If there are multiple things, they should each get their own property on this object.

Reply:
I wanted to keep the overall structure as close as possible to what the view was going to use to keep it as simple as possible and as one to one.

Reply:
Also I did this so that we wouldn't have to search arrays for existing data every-time since we should just do a key lookup to quickly check. Storing the data in arrays makes the process of updating the view much harder.

Reply:
I wasn't saying to use arrays instead of objects, i was just saying that when the root object is indexable, it's not clear what that index is actually holding, so I would prefer it to exist in a named property instead indexing the root object. 

If this is the exact structure required for the vscode tree view, then I'm ok with leaving it as is...but at least add some comments on this to explain what exactly it's holding and why it is done this way.

Reply:
Ah, ok. I see what you'er saying. I think I will move it into an `occurrences` field as currently we index by file and then also internally index by line in the file. This way I can keep the location to iterate over in the view the same key name.

Reply:
Done